x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -970,4 +971,36 @@ static DeprecationIssue checkMaxLocalStorageNodesSetting(final Settings settings
             DeprecationIssue.Level.CRITICAL
         );
     }
+
+    static DeprecationIssue checkSamlNameIdFormatSetting(final Settings settings,
+                                                         final PluginsAndModules pluginsAndModules,
+                                                         final ClusterState clusterState,
+                                                         final XPackLicenseState licenseState) {
+        final String principalKeySuffix = ".attributes.principal";
+        List<String> detailsList =
+            PRINCIPAL_ATTRIBUTE.getAttribute().getAllConcreteSettings(settings).sorted(Comparator.comparing(Setting::getKey))
+                .map(concreteSamlPrincipalSetting -> {
+                    String concreteSamlPrincipalSettingKey = concreteSamlPrincipalSetting.getKey();
+                    int principalKeySuffixIndex = concreteSamlPrincipalSettingKey.indexOf(principalKeySuffix);
+                    if (principalKeySuffixIndex > 0) {
+                        String realm = concreteSamlPrincipalSettingKey.substring(0, principalKeySuffixIndex);
+                        String concreteNameIdFormatSettingKey = realm + ".nameid_format";
+                        if (settings.get(concreteNameIdFormatSettingKey) == null) {
+                            return String.format(Locale.ROOT, "no value for [%s] set in realm [%s]",
+                                concreteNameIdFormatSettingKey, realm);
+                        }
+                    }
+                    return null;
+                })
+                .filter(detail -> detail != null).collect(Collectors.toList());
+        if (detailsList.isEmpty()) {
+            return null;
+        } else {
+            String message = "if nameid_format is not explicitly set, the previous default of " +
```

Comment:
I think we either need some more text here or a section in the breaking release notes and link from here ( I think the latter gives us more flexibility ). I can open a PR for the docs and reference the logs here once we have a link

Reply:
Hi @jkakavas -- just confirming that this is still on your to-do list, and you're not waiting on anything from me.

Reply:
Thank you for the ping, you are right, this slipped my mind. I'll work on something today or tomorrow and get @lockewritesdocs to review.

Reply:
Here is it: https://github.com/elastic/elasticsearch/pull/77785

Reply:
Thanks. I'll go ahead and merge this, and update its URL along with several others in an upcoming PR.